ProcessLib/Output/Output.cpp

```diff
@@ -159,20 +172,68 @@ void Output::doOutputAlways(Process const& process,
         return;
 
     std::string const output_file_name =
-        _output_file_prefix + "_pcs_" + std::to_string(process_id) + "_ts_" +
-        std::to_string(timestep) + "_t_" + std::to_string(t) + ".vtu";
+        constructFileName(_output_file_prefix, process_id, timestep, t) +
+        ".vtu";
     std::string const output_file_path =
         BaseLib::joinPaths(_output_directory, output_file_name);
 
     DBUG("output to %s", output_file_path.c_str());
 
     ProcessData* process_data = findProcessData(process, process_id);
     process_data->pvd_file.addVTUFile(output_file_name, t);
-    INFO("[time] Output of timestep %d took %g s.", timestep,
-         time_output.elapsed());
 
     makeOutput(output_file_path, process.getMesh(), _output_file_compression,
                _output_file_data_mode);
+
+    for (auto const& mesh_output_name : _mesh_names_for_output)
+    {
+        if (_meshes[0]->getName() == mesh_output_name)
+            continue;
+        auto& mesh = *BaseLib::findElementOrError(
+            begin(_meshes), end(_meshes),
+            [&mesh_output_name](auto const& m) {
+                return m->getName() == mesh_output_name;
+            },
+            "Need mesh '" + mesh_output_name + "' for the output.");
+
+        std::vector<MeshLib::Node*> const& nodes = mesh.getNodes();
+        DBUG(
+            "Found %d nodes for output at mesh '%s'.",
+            nodes.size(), mesh.getName().c_str());
+
+        MeshLib::MeshSubset mesh_subset(mesh, nodes);
+        std::unique_ptr<NumLib::LocalToGlobalIndexMap> mesh_dof_table(
+            process.getDOFTable(process_id)
+                .deriveBoundaryConstrainedMap(std::move(mesh_subset)));
+
+        processOutputData(t, x, mesh, *mesh_dof_table,
+                          process.getProcessVariables(process_id),
+                          process.getSecondaryVariables(), false,
+                          process.getIntegrationPointWriter(), _process_output);
+
+        // For the staggered scheme for the coupling, only the last process,
+        // which gives the latest solution within a coupling loop, is allowed to
+        // make output.
+        if (!(process_id ==
+                  static_cast<int>(_process_to_process_data.size()) - 1 ||
+              process.isMonolithicSchemeUsed()))
+            return;
+
+        std::string const mesh_output_file_name =
+            constructFileName(mesh.getName(), process_id, timestep, t) + ".vtu";
+        std::string const mesh_output_file_path =
+            BaseLib::joinPaths(_output_directory, mesh_output_file_name);
+
+        DBUG("output to %s", mesh_output_file_path.c_str());
+
+        //ProcessData* process_data = findProcessData(process, process_id);
```

Comment:
Do you still need these two lines?

Reply:
Add TODO for pvd and then a note for merging the bulk and arbitrary mesh output